TypeCobol/Tools/Options-Config/TypeCobolConfiguration.cs

```diff
@@ -398,20 +399,38 @@ public static bool CanCreateFile(string path)
         }
 
         /// <summary>
-        /// CreateFormat method to get the format name.
+        /// Create a new DocumentFormat instance for input documents
+        /// according to the specified format and encoding.
         /// </summary>
-        /// <param name="encoding">string</param>
-        /// <param name="config">Config</param>
-        /// <returns>DocumentFormat</returns>
-        public static Compiler.DocumentFormat CreateFormat(string encoding, ref TypeCobolConfiguration config)
+        /// <param name="format">Recognized format values are "zos", "utf8", "rdz" (default is "rdz").</param>
+        /// <param name="inputEncoding">Recognized input encoding values are those supported by System.Text.Encoding.GetEncoding(string) method.
+        /// Only applied when used with "rdz" format, ignored otherwise.</param>
+        /// <returns>instance of DocumentFormat.</returns>
+        public static Compiler.DocumentFormat CreateFormat(string format, string inputEncoding)
         {
-            config.EncFormat = encoding;
+            switch (format?.ToLower())
+            {
+                case "zos":
+                    return DocumentFormat.ZOsReferenceFormat;
+                case "utf8":
+                    return DocumentFormat.FreeUTF8Format;
+            }
+
+            var documentFormat = DocumentFormat.RDZReferenceFormat;
+            if (inputEncoding != null)
+            {
+                try
+                {
+                    var substitutionEncoding = Encoding.GetEncoding(inputEncoding);
+                    documentFormat = new DocumentFormat(substitutionEncoding, documentFormat.EndOfLineDelimiter, documentFormat.FixedLineLength, documentFormat.ColumnsLayout);
+                }
+                catch
+                {
+                    //Could not find the desired encoding, fallback to default encoding of RDZ format (UTF-8).
```

Comment:
We should emit a Warning when a System.ArgumentException is raised because Encoding.GetEncoding failed, this not a normal situation.

Reply:
See latest commit.